regparser/web/jobs/views.py

```diff
@@ -91,21 +110,48 @@ def create(self, request, *args, **kwargs):
         return Response(serialized.data, status=status.HTTP_201_CREATED,
                         headers=headers)
 
+
+class JobViewList(mixins.ListModelMixin,
+                  mixins.CreateModelMixin,
+                  generics.GenericAPIView):
+    """
+    Handles the list view for jobs of all types.
+    Should be subclassed along with ``BaseViewList`` for classes handling
+    specific job types.
+    """
+    queryset = PipelineJob.objects.all()
+    renderer_classes = renderer_classes
+    serializer_class = PipelineJobSerializer
+
+    def filter_queryset(self, request, *args, **kwargs):
+        """
+        Overridden in order to get data from the Redis queue as well as the DB.
+
+        Impure
+            Pulls information from the DB and the Redis queue.
+
+        :arg HttpRequest request: the incoming request.
+
+        :rtype: list[PipelineJob]
+        :returns: List of PipelineJob objects.
+        """
+        queryset = super(JobViewList, self).filter_queryset(request, *args,
+                                                            **kwargs)
+        queryset = add_redis_data_to_job_data(queryset)
+        return queryset
```

Comment:
That's an excellent point; I should investigate this further. Right now, I'm not actually forcing the `queryset` into another form, just going through it in a `for` loop and adding information to each item, so I *think* it actually is still a `queryset`. But the docstrings for this and `add_redis_data_to_job_data` are wrong in asserting that `list`s are what's being handled.